cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +69,22 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def find_non_dev(non_dev_deps, yarn_lock, current_dep):
+    if current_dep not in non_dev_deps:
+        non_dev_deps.append(current_dep)
+        sub_deps = []
+        for dep_identifier, dep_data in yarn_lock.items():
+            if current_dep in dep_identifier:
+                if "dependencies" in dep_data:
+                    sub_deps = [
+                        f"{name}@{version}" for name, version in dep_data["dependencies"].items()
+                    ]
+                break
+        for sub_dep in sub_deps:
+            find_non_dev(non_dev_deps, yarn_lock, sub_dep)
```

Comment:
Can you please elaborate on why you choose finally DFS? 

I don't think there is a high probability of hitting the recursion limit with 1000+ nested deps, but maybe it's better to avoid future problems

Reply:
Let's use BFS with collections.deque instead of DFS

Reply:
I've feeling that DFS may be faster and take less resources than BFS, no hard data though, I dunno how common dependencies looks like

Reply:
Apologies, I actually have implemented BFS here, @akhmanova . I accidentally wrote DFS instead of BFS in my daily standup. The reason for BFS would be I have noticed that most dependencies have a more immediate nested dependencies (many dependencies have a double-digit number of sub-dependencies) than the depth of how far those dependencies go (usually ~2-3). I will change the algorithm to use collections.deque as it should help with performance. I will also change the function from a recursive one to an iterative one, @MartinBasti

Reply:
I may need explanation how this is BFS, because I see well hidden DFS

Reply:
I believe the old code wasn't designed as intended and caused some confusions. The new changes I have made now should be a clear BFS algorithm.